packages/nodes-base/nodes/Asana/GenericFunctions.ts

```diff
@@ -29,10 +29,16 @@ export async function asanaApiRequest(
 		method,
 		body: method === 'GET' || method === 'HEAD' || method === 'DELETE' ? null : { data: body },
 		qs: query,
-		url: uri || `https://app.asana.com/api/1.0${endpoint}`,
+		url:
+			uri ||
+			`https://app.asana.com/api/1.0/${endpoint.startsWith('/') ? endpoint.slice(1) : endpoint}`,
```

Comment:
we should instead change the `endpoint` type from `string` to `/${string}`, and that way this is caught at compile-time, and we don't need to maintain yet another line that conditionally removes `/`.

Reply:
I've pushed my proposed changes [here](https://github.com/n8n-io/n8n/compare/gh/8403...8404-proposal?expand=1). If they look acceptable, please feel free to cherry-pick.

Reply:
Checked the URIs and it looks like they are all ok anyway, I was just being cautious based on previous experience. Also spotted a typo and fixed that.

Reply:
Ah it didn't update quick enough to show your changes, Will grab your changes now as they make more sense.